src/VisualStudio/Core/SolutionExplorerShim/HierarchyItemToProjectIdMap.cs

```diff
@@ -42,7 +43,9 @@ public bool TryGetProjectId(IVsHierarchyItem hierarchyItem, string targetFramewo
                     .Where(p =>
                     {
                         if (p.Hierarchy.TryGetCanonicalName((uint)VSConstants.VSITEMID.Root, out string projectCanonicalName)
-                            && projectCanonicalName.Equals(nestedCanonicalName, System.StringComparison.OrdinalIgnoreCase))
+                            && p.Hierarchy.TryGetItemName((uint)VSConstants.VSITEMID.Root, out string projectName)
+                            && projectCanonicalName.Equals(nestedCanonicalName, System.StringComparison.OrdinalIgnoreCase)
+                            && projectName.Equals(nestedName))
```

Comment:
Or at least document what you are using?

Isn't there a hierarchy property that is the full path to the project file that you could just use instead?